test/cockroachdb_test.go

```diff
@@ -0,0 +1,82 @@
+package helm_testing
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/require"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+func Test_Role(t *testing.T) {
+	helmChartPath, err := filepath.Abs("../cockroachdb")
+	require.NoError(t, err)
+	require.NotNil(t,helmChartPath)
+
+	t.Run("Test TLS Enabled but certs not provided", func(t *testing.T) {
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"tls.enabled": "True",
+				"tls.certs.provided": "False",
+				"tls.certs.certManager": "False",
+			},
+			KubectlOptions: k8s.NewKubectlOptions("", "", "test-deployment"),
+		}
+
+		releaseName := "cockroachdb-test"
+
+		output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{"templates/role.yaml"})
+		require.NotNil(t, output)
+
+		var role rbacv1.Role
+		helm.UnmarshalK8SYaml(t, output, &role)
+
+		verbs := role.Rules[0].Verbs
+
+		require.Len(t, verbs, 2)
+		require.Contains(t, verbs, "get")
+		require.Contains(t, verbs, "create")
+	})
+
+	t.Run("Test TLS Enabled and certs provided", func(t *testing.T) {
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"tls.enabled": "True",
+				"tls.certs.provided": "True",
+				"tls.certs.certManager": "False",
+			},
+			KubectlOptions: k8s.NewKubectlOptions("", "", "test-deployment"),
+		}
+
+		releaseName := "cockroachdb-test"
+
+		output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{"templates/role.yaml"})
+		require.NotNil(t, output)
+
+		var role rbacv1.Role
+		helm.UnmarshalK8SYaml(t, output, &role)
+
+		verbs := role.Rules[0].Verbs
+
+		require.Len(t, verbs, 1)
+		require.Contains(t, verbs, "get")
+	})
+
+	t.Run("Test TLS disabled", func(t *testing.T) {
+		options := &helm.Options{
+			SetValues: map[string]string{
+				"tls.enabled": "False",
+				"tls.certs.provided": "True",
```

Comment:
if you're testing TLS disabled, this line should also be false, and the db should be running in insecure mode

Reply:
The DB will run in insecure mode when tls.enabled is false?

https://github.com/cockroachdb/helm-charts/blob/f37ff112fbbee969d64dc19526226c62bb9651fc/cockroachdb/templates/statefulset.yaml#L224

This is something that a test on the statefulset template could ensure.